src/main/java/Duke.java

```diff
@@ -1,10 +1,129 @@
+                System.out.println("Congrats! You've completed this task: [X] "
+                        + tasks[num].getTaskDesc());
+            } catch (NumberFormatException e) {
+                System.out.println(SOMETHING_WRONG +
+                        "Please ensure that you have entered the index of the task(s) to be marked as done");
+                break;
+            }
+        }
+    }
+
+    private static void listAllTasks(Task[] tasks) {
+        for (int i = 0; i < nextAdd; i++) {
+            StringBuilder sb = new StringBuilder();
+            String done = tasks[i].isPending() ? "[ ]" : "[X]";
+            sb.append(i + 1).append(": ").append(tasks[i].getType())
+                    .append(done).append(" ").append(tasks[i].getTaskDesc());
+            if (!tasks[i].getAdditionalInfo().isEmpty()) {
+                sb.append(" ").append(tasks[i].getAdditionalInfo());
+            }
+            System.out.println(sb);
+        }
+    }
+
+    private static void tasksAddOrModify(String input, Task[] tasks) {
+
+        String[] inputWords = input.split(" ");
+        if (inputWords[0].equals("done")) {
+            makeTaskDone(inputWords, tasks);
+        } else {
+            if (inputWords[0].equals("deadline")) {
+                int num = Integer.parseInt(inputWords[1]) - 1;
+                Deadline temp = new Deadline(tasks[num].getTaskDesc());
+                String deadlineString = createRemainingString(inputWords);
+                temp.setByDateTime(deadlineString);
+                tasks[num] = temp;
+                System.out.println("Deadline set: " + deadlineString);
+            } else if (inputWords[0].equals("event")) {
+                int num = Integer.parseInt(inputWords[1]) - 1;
+                Event temp = new Event(tasks[num].getTaskDesc());
+                temp.setAtDateTime(createRemainingString(inputWords));
+                tasks[num] = temp;
+                System.out.println("Event set");
+            } else {
+                Todo temp = new Todo(input);
+                tasks[nextAdd] = temp;
+                System.out.println("Added: " + input);
+                nextAdd++;
+            }
+        }
+
+    }
+
+
+    private static void userCommands() {
+
+        boolean isProgramRunning = true;
+        Scanner scanner = new Scanner(System.in);
+        Task[] tasks = new Task[100];
+
+        while (isProgramRunning) {
+            System.out.print("Enter your command here: ");
+            String input = scanner.nextLine();
+            if (input.isEmpty()) {
+                //TODO: print something wrong
+            } else if (input.equals("help")) {
+                //TODO: print help statement
+            } else if (input.equals("bye")) {
+                isProgramRunning = false;
+            } else if (input.equals("list")) {
+                listAllTasks(tasks);
+            } else {
+                tasksAddOrModify(input, tasks);
+            }
```

Comment:
not necessarily a violation, but you could consider using a switch statement for better readability. You could also consider using constant variables for your keywords, ie. help, bye, deadline, todo, etc.